### PR TITLE
fixes safari browser missing buffer when using webaudio

### LINKED ISSUE #952: How can I play one song and have 2 waveforms in sync? & safari render issue
In an app i'm making I need to show the waveform twice in separate pieces of ui. These would need to be in sync, is this possible? I see you allow split channels which is close to what i want but in my case it would be a straight duplicate.

Thanks,

### PR DESCRIPTION
**Pull request by entonbiba**
fixes #952 safari buffer undefined

```
var newBuffer = this.createBuffer(1, 4096, this.sampleRate);
```
- it creates the missing buffer based on 1 channel, 4096 samples and the sampleRate from the current webaudio context.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
why are these values (1 channel, 4095 samples) hard coded? I guess Safari doesn't supply these values somewhere?

**Comment by entonbiba** [comment]
@thijstriemstra yea that's the problem it doesn't. Safari TP should return the correct buffer, but it's a long way until that will be available. For now that's the best we can do to fix it in safari.

File: src/webaudio.js [hunk 1]
```diff
@@ -223,6 +223,11 @@ WaveSurfer.WebAudio = {
 
         this.setLength(length);
 
+        if (!this.buffer.length) {
```
Comment (katspaugh) [inline src/webaudio.js]: Could you add a short comment regarding the Safari quirk?

Comment (thijstriemstra) [inline src/webaudio.js]: ping @entonbiba

Comment (entonbiba) [inline src/webaudio.js]: @katspaugh sure

Comment (entonbiba) [inline src/webaudio.js]: @thijstriemstra @katspaugh done

**Comment by katspaugh** [comment]
Thank you @entonbiba!

**Comment by thijstriemstra** [comment]
> yea that's the problem it doesn't. Safari TP should return the correct buffer, but it's a long way until that will be available. 

@entonbiba can this workaround be removed yet?

**Pull request merged**